src/server/sql/reading/create_compressed_reading_views.sql

```diff
@@ -149,31 +155,60 @@ CREATE INDEX if not exists idx_daily_readings ON daily_readings USING GIST(time_
-		number would lead to really large scans of the hourly table (which is not materialized) right
-		before the switch to days.
+		The minimum day and hourly points are determined so that data is read from the daily view when the
+		requested interval is at least two months, from the hourly view when between two months and at least two weeks,
+		and directly from the raw readings table when under two weeks.
+
+		+----------------------+--------------------
+		| Space considerations |
+		+----------------------+
+		Each row returned by this function contains 1 INT (meter_id), 2 TIMESTAMPs (start/end stimestamp), and
+		1 FLOAT (reading_rate). According to PostgreSQL documentation, the storage size of INT is 4 bytes,
+		TIMESTAMP is 8 bytes, and FLOAT is 8 bytes, and so each row totals to 28 bytes of data.
+
+		For example, if the requested range is one year, this algorithm will read from the materialized daily view
+		and return 0.01 megabytes of data.
+		
+		From the hourly view, the number rows that can be read ranges from 360 (the minimum) to 60 * 24 = 1440
+		points. Hence, the amount of data read from the hourly view ranges from 0.010 to 0.040 megabytes
+		per meter. 
+		
+		From the readings table, the number of rows returned depends on the raw reading granularity.
+		The most common reading granularity for electric meters is 15-minutes, which means the meter
+		stores data for every 15-minute interval.
+
+		The transition from hourly to raw readings occurs when the number of hourly points is 359 or less which is
+		less than 15 days or roughly two weeks.
 
-		For example, if we specified at least 100 points, and the range was 99 days and 23 hours, then we would have
-		to scan (100 days * 24 hours - 1 hour) * (60 minutes) * (n meters) worth of data. That's 143940 rows per meter
-		if the meters are minute precision! Each meter reading has at least 4 ints (meter_id, reading, start/end timestamp),
-		totalling slightly over 2.2 megabytes of data per meter. This balloons significantly when it's sent as JSON,
-		leading to unacceptable load times for the client, especially if they're on a mobile device.
+		If the granularity is 15-minutes, then the maximum amount of raw points is 359 * 4 = 1436, which translates
+		to 0.040 megabytes per meter.
+		If instead the raw readings granularity is 1-minute, then the maximum amount of raw points is 359 * 60 = 21540,
+		which translates to 0.60 megabytes per meter. 
+		
+
+		+------+-------------------
+		| TODO |
+		+------+
+		- Note that the above documentation does not account for the increase in data when packaged in JSON format.
+		- This function is complicated by the fact that groups can contain meters with different levels of granularity.
+		We address this now by having sites pick a site-level reading frequency. (Not done)
+		- When resource generalization is implemented in OED, this function will need to be done in a 
```

Comment:
Yes, by "different levels of granularity" I mean different reading rates. I was using the language that I see in the eGauge documentation. I mainly pointed out resource generalization because different resource meters will likely have different reading rates, but you're correct that meters with different reading rates is not unique to resource generalization. I might have just been over-thinking this comment. I have updated the comment.